runtime/onert/core/src/util/EventRecorder.h

```diff
@@ -17,28 +17,75 @@
 #ifndef __ONERT_UTIL_EVENT_RECORDER_H__
 #define __ONERT_UTIL_EVENT_RECORDER_H__
 
+#include "util/TracingCtx.h"
+
 #include <map>
 #include <memory>
 #include <mutex>
 
 #include <vector>
 
+struct SubgDurationEvent;
+struct OpDurationEvent;
+
+class DurationEventVisitor
+{
+public:
+  virtual ~DurationEventVisitor() = default;
+
+  virtual std::string visit(const SubgDurationEvent &) const
+  {
+    throw std::runtime_error("Please implement");
+  }
+  virtual std::string visit(const OpDurationEvent &) const
+  {
+    throw std::runtime_error("Please implement");
+  }
+};
```

Comment:
(Opinion-based) IMHO introducing a visitor for DurationEvent is an overkill since:

- There are only two events
- What visitors do are relatively simple
- Changing event-recording module does not affect other modules

I'm not saying it is unacceptable, though.

Reply:
I will change visitors to two class architecture.